### PR TITLE
Basic tree node label editing

### PR DESCRIPTION
**Pull request by rad-pat**
I've had the need to edit the nodes in a tree and wondered if this might be useful to have as an option in the main source. It seemed that the basic edit wasn't too much trouble so I wondered why it was not included in the first instance? Perhaps there wasn't the requirement. The node editor works ok as a cell editor, but could do with being aligned with the node in the tree and I wasn't sure of the best way to go about that. Any pointers would be really helpful and I'll get this fixed up.

### CODE REVIEW

**Comment by cboulanger** [comment]
This would be a great addition to the core. We just have to make sure that it doesn't break anything. Unfortunately, ther are no tests in the framework that test the behavior of the TreeVirtual widget, but could you maybe add a test case to https://github.com/qooxdoo/qxl.demobrowser/tree/master/source/demo/treevirtual using your fork, which could then be added to the demobrowser once the PR has been accepted?

**Comment by cboulanger** [comment]
I'll try to test the PR branch with my app in the next few days to see how it works.

**Comment by rad-pat** [comment]
Thanks for looking, yep I'll add some demo code

File: framework/source/class/qx/ui/treevirtual/SimpleTreeDataModel.js [hunk 1]
```diff
@@ -244,7 +244,7 @@ qx.Class.define("qx.ui.treevirtual.SimpleTreeDataModel",
       // The tree column is not editable
```
Comment (derrell) [inline framework/source/class/qx/ui/treevirtual/SimpleTreeDataModel.js:244]: This comment is no longer accurate, it seems.

File: framework/source/class/qx/ui/treevirtual/celleditor/NodeEditor.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    },
+
+
+    // interface implementation
+    createCellEditor : function(cellInfo)
+    {
+      var cellEditor = this._createEditor();
+
+      // The value in the case of a Tree is a Node and we want the label
+      if (cellInfo.value === null || cellInfo.value === undefined) {
```
Comment (derrell) [inline framework/source/class/qx/ui/treevirtual/celleditor/NodeEditor.js:52]: If `cellInfo.value` is required to be an object, as indicated by the value set below, then this test might better be
```
if (cellInfo.value === null || typeof cellInfo.value != "object")
```

**Review by derrell — CHANGES_REQUESTED** [review]
This looks like a good addition.

**Review by derrell — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]
Couldn't test it yet, but it looks good to me.

**Review by hkollmann — APPROVED** [review]

**Comment by cboulanger** [comment]
Thank you @rad-pat for this addition!

**Pull request merged**